src/rt/sched/schedulerthread.h

```diff
@@ -168,6 +187,131 @@ namespace verona::rt
       }
     }
 
+    /**
+     * Mute a set of cowns. This will add the cowns to the mute set of the
+     * mutor.
+     */
+    void mute(T** cowns, size_t count)
+    {
+      auto it = mute_map.find(mutor);
+      if (it == mute_map.end())
+      {
+        auto* set = ObjectMap<T*>::create(alloc);
+        it = mute_map.insert(alloc, std::make_pair(mutor, set)).second;
+      }
+      else
+      {
+        mutor->weak_release(alloc);
+      }
+      auto& mute_set = *it.value();
+
+      for (size_t i = 0; i < count; i++)
+      {
+        auto* cown = cowns[i];
+
+        if (
+          (state == ThreadState::PreScan) || (state == ThreadState::Scan) ||
+          (state == ThreadState::AllInScan))
+        { // Messages in this cowns queue must be scanned.
+          cown->schedule();
+          continue;
+        }
+
+        Systematic::cout() << "Mute " << cown << std::endl;
+        auto bp = cown->backpressure.load(std::memory_order_relaxed);
+        bp.set_muted();
+        cown->backpressure.store(bp, std::memory_order_release);
+        T::acquire(cown);
+        mute_set.insert(alloc, cown);
+      }
+
+      alloc->dealloc(cowns, count * sizeof(T*));
+    }
+
+    /**
+     * Unmute all mute sets where the mutor is in a state that triggers
+     * unmuting. If `force` is true, then all mute sets in the map will be
+     * unmuted.
+     */
+    template<bool force = false>
```

Comment:
Why does `force` need to be a template parameter and not a function argument?

Reply:
The thought was to avoid a branch in the case where force is true without requiring the function to be inlined. But perhaps that is not worth the syntax of a template parameter?

Reply:
I'd leave it without template until we figure that loop is super hot and we need to optimise for a case that we know is the vast majority. Right now, it's not clear. I'm not expecting the `mute_map` to be so big as to take considerable computation resources comparable to the threads actual load.

Reply:
Ok, fixed.